src/wasm/wasm-type.cpp

```diff
@@ -466,6 +472,11 @@ bool isTemp(HeapType type) {
   return !type.isBasic() && getHeapTypeInfo(type)->isTemp;
 }
 
+bool isBasicOrBasicKind(HeapType type) {
+  return type.isBasic() ||
+         getHeapTypeInfo(type)->kind == HeapTypeInfo::BasicKind;
+}
```

Comment:
In most situations, yes. However, in `globallyCanonicalize`, we need to replace BasicKind HeapTypes with their underlying Basic HeapTypes because BasicKind HeapTypes should never be exposed to the user. So that particular piece of canonicalization code needs to be able to differentiate them, but nothing else does. This new function is for helping all the other code ignore the difference.

Reply:
I see, thanks. Is there already a comment with that elsewhere? If not then maybe add it here?